examples/pipelines/datacomp/components/detect_text/src/main.py

```diff
@@ -0,0 +1,194 @@
+             to an 1-D numpy array. E.g. list[float], np.ndarray,
+             or torch.Tensor. Polygon is written in
+             [x1, y1, x2, y2, ...].
+
+     Returns:
+         np.array: The converted bounding box [x1, y1, x2, y2]
+    """
+    if len(polygon) % 2 != 0:
+        raise ValueError("Not ok")
+    polygon = np.array(polygon, dtype=np.float32)
+    x = polygon[::2]
+    y = polygon[1::2]
+    return np.array([min(x), min(y), max(x), max(y)])
+
+
+def process_image(image: bytes, *, image_transform, device: str) -> torch.Tensor:
+    """
+    Process the image to a tensor.
+
+    Args:
+        image: The input image as a byte string.
+        image_transform: The image transformation to apply.
+        device: The device to move the transformed image to.
+    """
+
+    def load(img: bytes) -> Image:
+        """Load the bytestring as an image."""
+        bytes_ = io.BytesIO(img)
+        return Image.open(bytes_).convert("RGB")
+
+    def transform(img: Image) -> torch.Tensor:
+        """Transform the image to a tensor and move it to the specified device."""
+        return image_transform(img)[0][0].to(device)
+
+    return transform(load(image))
+
+
+def detect_text_batch(
+    image_batch: pd.DataFrame,
+    *,
+    cfg,
+    model,
+) -> pd.Series:
+    """Detext text on a batch of images."""
+    imgs = torch.stack(list(image_batch), dim=0)
+    batch_size = imgs.shape[0]
+    # TODO fix this, make this component also take width and height
+    # as input in spec to process arbitrarly sized images
+    img_metas = {
+        "filename": [None for i in range(batch_size)],
+        "org_img_size": torch.ones((batch_size, 2)).long() * 256,
+        "img_size": torch.ones((batch_size, 2)).long() * 512,
+    }
+
+    data = dict()
+    data["imgs"] = imgs
+    data["img_metas"] = img_metas
+    data.update(dict(cfg=cfg))
+
+    # forward
+    with torch.no_grad():
+        outputs = model(**data)
+
+    # get cropped images
+    boxes_batch = []
+    for i in range(batch_size):
+        raw_contours = outputs["results"][i]["bboxes"]
+
+        boxes = []
+        for polygon in raw_contours:
+            box = poly2bbox(polygon)
+            boxes.append(box)
+        boxes_batch.append(boxes)
+
+    return pd.Series(boxes_batch, index=image_batch.index)
+
+
+class DetectTextComponent(PandasTransformComponent):
+    """Component that detects text in images using an mmocr model."""
+
+    def __init__(self, *args, batch_size: int) -> None:
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        logger.info(f"Device: {self.device}")
+
+        self.batch_size = batch_size
+        self.image_transform = SquarePadResizeNorm(
+            img_size=512,
+            norm_mean=(0.485, 0.456, 0.406),
+            norm_std=(0.229, 0.224, 0.225),
+        )
+
+        cfg = Config.fromfile("config/fast/tt/fast_tiny_tt_512_finetune_ic17mlt.py")
```

Comment:
I would remove all the other configs for the other models. It's a lot of files to upload. We can maybe think about developing the model later to be more generic then you should include the model name as a component argument